### DISCUSSION TITLE
Discriminate union based on a different field in the parent model

### DISCUSSION THREAD

**Discussion by nextmat** [pydantic/pydantic#10388] [Question]
With a model like this:

```py
class Connection(BaseModel):
    protocol: str
    protocol_params: Union[ProtocolOneParam, ProtocolTwoParam, ...]
    # more fields...
```

Is it possible to discriminate the validation type for `protocol_params` based on the value of `protocol`?

I know Pydantic v2 added [callable discriminators](https://docs.pydantic.dev/latest/concepts/unions/#discriminated-unions-with-callable-discriminator). However these seem to only include the value of the field that is being discriminated on, so I don't think they work for this case?

**Comment by tyteen4a03** [comment]
+1 - have the same question

**Comment by Abdelhadi92** [comment]
I have the same question

**Comment by evanobrientt** [comment]
I also have this question

**Comment by jvaesteves** [comment]
I came across this discussion after I posted mine 🤦‍♂️, but in case anyone else happens to arrive here, I implemented [a way to achieve mostly the same](https://github.com/pydantic/pydantic/discussions/11795#discussioncomment-12946885) (I haven't tested it thoroughly yet though)
